src/ompl/geometric/planners/rrt/src/RRTConnect.cpp

```diff
@@ -156,7 +156,10 @@ ompl::geometric::RRTConnect::GrowState ompl::geometric::RRTConnect::growTree(Tre
         if (si_->getMotionStates(astate, bstate, states, count, true, true))
             si_->freeState(states[0]);
 
-        for (std::size_t i = 1; i < states.size(); ++i)
+        // Add states forwards if from start, backwards if from goal
+        for (std::size_t i = tgi.start ? 1 : states.size() - 1;  //
+             (tgi.start) ? i < states.size() : i > 0;            //
+             i += 2 * tgi.start - 1)
```

Comment:
That seems rather obfuscated. Since `i` isn't actually used, wouldn't this be cleaner?:
```
            for (const auto& state : tgi.start ? states : (states | std::views::reverse))
            {
                ...
                motion->state = state;
                ...
            }
```
This requires C++20, but this can probably also be done in C++17 by using range-v3 or boost.

Reply:
I changed up the implementation to use something more intuitive than arcane indexes, not as slick as the suggested but I think will be clear.